scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -117,7 +117,40 @@ interface ScanResultsStorage {
      *
      * @return The [ScanResultContainer] matching the [id][Package.id] of [pkg] and the [scannerDetails].
      */
-    fun read(pkg: Package, scannerDetails: ScannerDetails): ScanResultContainer
+    fun read(pkg: Package, scannerDetails: ScannerDetails): ScanResultContainer {
```

Comment:
whole commit: Instead of putting more code into the interface I suggest to instead
clean up the interface from implementations completely. The implementation could be
either 
(1) moved into a new maybe abstract base class (if inheritance is really needed) or 
(2) moved into a utility class used via aggregation

Both approaches would allow re-use (which is as I understand the goal of this commit) 
while being a bit cleaner in terms of encapsulation and coupling. 
Amongst others it would allow e.g. moving the access statistic function signature to the interface
and make it easier to get rid of this companion object which in the end is a singleton and thus at least is a code smell.
Also 'patchScanCodeLicenseRefs' would not need to be exposed via the interface.

Reply:
I hope to have addressed (most) of your concerns by not moving code to the interface, but to a new abstract class for file-based storages, which is what I had discussed with @mnonnenmacher in a meeting. I'm keeping the companion object for now to reduced changes to a minimum for introducing the `LocalFileStorage` while not interfering too much with @mnonnenmacher's Postgres-storage work. That work will anyway require a few more refactorings, and we can hopefully address any remaining concerns as part of that work.